Data_Generator/config.txt
#-------------------------------------------------------------
# 文件
device_data_path = /mnt/e/Files/Code/cross_plateform/C_C++/code_new/Data/input/selectTxt_unix_original_for_new_code.txt
mole_device_data_path = /mnt/e/Files/Code/cross_plateform/C_C++/code_new/Data/input/selectTxt_SAM_mole_filted.txt
# mole_onehot_data_path = /mnt/e/Files/Code/cross_plateform/C_C++/code_new/Data/input/molecule_collection_encoded_filted.txt
mole_onehot_data_path = /mnt/e/Files/Code/cross_plateform/C_C++/code_new/Data/input/molecule_collection_descritpor_filted.txt

predict_data_read_path = /mnt/e/Files/Code/cross_plateform/C_C++/code_new/Data/input/selectTxt_unix_original_for_new_code.txt
predict_data_saving_path = /mnt/e/Files/Code/cross_plateform/C_C++/code_new/Data/output/predict_data.txt
# maps存放文件夹
map_file_root = /mnt/e/Files/Code/cross_plateform/C_C++/code_new/Data/maps 
# 输出数据文件夹
output_dir = /mnt/e/Files/Code/cross_plateform/C_C++/code_new/Data/output
# 存储onehot数据的文件夹
onehot_dir_root = /mnt/e/Files/Code/cross_plateform/C_C++/code_new/Data/onehot_data
# 23列分子符号文件夹
col_23_config_file_path = /mnt/e/Files/Code/cross_plateform/C_C++/code_new/Data/config/col_23_words.txt 

#-------------------------------------------------------------
# 是否使用已经建构的maps，0 代表不用，则重新生成maps到指定目录 1 代表使用
using_exist_maps = 1 

# 处理分子/电池数据 0 处理SAMs数据 1 处理电池数据 
device_data = 0

# 数据保留尺度
device_data_reserve_ratio = 1

# ------------------------------------------------------------
# 列种类
sing_col = 0,1,2,4,5,6,7,8,9,10,14,15,16,17,18,19,21,22,24,25,27,29,35,38,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63
sing_multi_col = 13,20,23,26,28,30,31,41,43,44
hybrid_multi_col = 3,11,12,32,33,34,36,37,39,40,42
full_structure_col = 3      # 本列包含完整的器件结构，但本项目不使用此列，而是通过子结构进行组合

num_col = 1,4,5,10,20,26,29,34,36,37,43,49,51,52,53,54,55,56,57,58,59
JV_col = 52,53,54,55,56,57,58,59
# col 23 指定钙钛矿组成表达式是哪个列
expression_col = 23

# 序列组
sequence_block = [[11],[12,13],[30,31,32,33,34],[36,37],[39,41],[42,43,44]]

# col 2 (years col) 指定年文件夹是哪一列
years_col = 1

tl_layer_transparent = 12
tl_layer_metal_Pole = 39
transparent_pole_layer = 11
metal_pole_layer = 42
#-------------------------------------------------------------
# 预先过滤数据的JV参数范围
Voc_range = 0,10
Isc_range = 0,30
FF_range = 0,1
PCE_range = 0,30
# JV 合理性范围（即Isc*Voc*FF=PCE的误差）
jv_accuracy = 0.05
#-------------------------------------------------------------
# maps大小，即每一列填充数据长度（分隔符| >>分隔的单列填充长度）
maps_size = 20,0,200,300,0,0,7,2,2,2,0,40,767,51,2,3,3,3,3,0,3,3,3,90,3,3,0,3,9,0,34,4,26,38,0,3,0,0,0,900,54,38,46,0,23,3,3,3,3,0,3,0,0,0,0,0,0,0,0,1,1,1,1,2,500,500
block_size = 1,1,1,3,1,1,1,1,1,1,1,3,3,3,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,3,3,3,1,3,3,1,3,3,3,3,3,3,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1
# 数字填充长度
num_col_sig_col_size = 0,0,0,0,1,1,0,0,0,0,1,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,1,0,0,1,0,0,0,0,4,0,4,4,0,0,0,0,0,1,0,0,0,0,0,1,0,1,1,1,1,1,1,1,1,1,0,0,0,0,0,0
# ------------------------------------------------------------
# 连续数据，用于数据生成
data_type = [[0,1,2,3,49,50,51],[4],[5],[6],[7],[8],[9],[10],[11],[12],[13],[14,15,16,17,18,19,20,21,22,23,24,25,26,27,28],[29,30,31,32,33,34],[35,36,37],[38],[39,40,41],[42,43,44],[45,46,47,48],[52],[53],[54],[55],[56],[57],[58],[59],[60],[61],[62],[63]]


Fitting_Model/params.py
# 弱分类器 DecisionTreeClassifier 和 MLP
ada_param = [
    [0],                      # 弱分类器种类：决策树/mlp(不支持sampleWeight，后期再写)
    [50],                       # 估计器数量
    # ada参数
    ['SAMME'],                  # algorithm SAMME.R为回归模型参数，未来将被舍弃
    [0.1,0.3,0.5,1,1.2,1.5],    # leanring rate
    # DecisionTreeClassifier超参数
    ['gini'],                       # criterion (4)
    ['random', 'best'],             # splitter
    [2, 5, 10, 15],                   # min_samples_split
    [1, 5, 10],                      #
    ['balanced', 'None'],           # class_weight
    #mlp 超参数（参考MLP一章）
    [0, 50, 100],
    [0, 50, 100],                         # layers
    ['identity', 'logistic', 'tanh','relu'],
    ['lbfgs','sgd','adam'],
    [0,0.0001,0.0002,0.0004,0.0008],    # alpha（先用0）
    ['adaptive'],                       # learning rate
    [0.7, 0.9, 0.95, 0.97, 0.98, 0.99],            # monument（先用默认0.9）
]

# Gaussian 无超参数
# Multi 无超参数
# Cata 有一个bool型
# comple 有两个
BO_param = [
        [0,1,2,3],                  # BO type
        # Comple
        [0.001,0.01,0.05, 0.1,0.3, 0.5, 0.7, 1],       # alpha
        ['True', 'False'],        # norm
        ]

# KNN网格参数采样范围
'''https://blog.csdn.net/weixin_42182448/article/details/88639391'''
knn_param = [
    [3,4,5,6,7],                                            # n_neighbors 近邻邻居
    ['uniform','distance'],                                 # weights 距离权重
    ['ball_tree', 'kd_tree'],                               # algorithm 距离权重
    [30,50,70,100],                                         # leaf_size kd/ball树才有
    ['euclidean', 'minkowski','manhattan','chebyshev'],     # metric https://blog.csdn.net/weixin_44607126/article/details/102598096
    [1,2]                                                   # p(用于minkowski) 1 曼哈顿距离 2 欧
]

# 构建网格筛选器
lr_param = [
    # 公共参数
    [0,1,2],    # 0 elastic 1 lasso 2 ridge
    [0.01,0.05,0.1,0.3,0.5,1,1.5,2,5,10],       # alpha
    # elastic 和 lasso
    ['cyclic', 'random'],                       # selection
    # elastic
    [0.2,0.3,0.4,0.5,0.6,0.7,0.8],              # l1_ratio(3)
    # ridge
    ['balanced', 'None'],                       # class_weight(4)
    ['auto']                                    # solver
]

# 作为弱分类器的参数
mlp_sk_weak_list = [
    [0,50,100],
    [0,50,100],                         # layers
    ['identity', 'logistic', 'tanh','relu'],
    ['lbfgs','sgd','adam'],
    [0,0.0001,0.0002,0.0004,0.0008],    # alpha（先用0）
    ['adaptive'],                       # learning rate
    [0.7, 0.9, 0.95, 0.97, 0.98, 0.99],            # monument（先用默认0.9）
]

# 作为普通（强）分类器的参数 最高六层
mlp_param_list = [
    [0,100,300,500],
    [0,100,300,500],
    [0,100,300,500],
    [0,100,300,500],
    [0,100,300,500],     
    [0,100,300,500],# hidden_layer_size（最大六层）
    ['identity', 'logistic', 'tanh','relu'],         #activation
    ['libfgs', 'sgd', 'adam'],  # solver
    [0,0.0001,0.001,0.01,0.1],  # l2 punished
    ['auto'], # batch_size(使用默认)
    ['adaptive'], # learn_rate（）
    [0.001], # learn_rate_init 使用默认值
    [0.0001,0.0005,0.001], # power_t
    [0.7, 0.8, 0.9, 0.95], #mometum
    [], # nestervous_momentum
    [0.1], # validation_fraction
    [],  #beta 1
    [], #beta 2
]

# https://blog.csdn.net/VariableX/article/details/107190275
# https://blog.csdn.net/yingfengfeixiang/article/details/79369059
# 默认尺寸：50
rf_param = [
    ['gini', 'entropy', 'log_loss'],
    [2,6,12,24],                    # max_depth
    [1,2,5,10],                     # min_samples_leaf 
    [2,5,10,15,100],              # min_samples_split
    ['log2','sqrt'],                # max_features
    ['None', 'balanced']
]

# 仅使用linearSVC
# https://zhuanlan.zhihu.com/p/341554415
# https://blog.csdn.net/weixin_41990278/article/details/93165330
# https://blog.csdn.net/qq_41264055/article/details/133121451
# default '0'
svm_param = [
    # Public
    ['linear', 'poly', 'rbf', 'sigmoid'],
    #------------------------------
    # LinearSVC(linear)
    ['l1','l2'],                                    # penalty(1)
    [0],                                            # C（见7）
    ['hinge', 'squared_hinge'], # loss
    [True, False],                          # dual（优先 false）
    ['ovr', 'crammer_singer'],                      # multi_class 
    ['None'],                                       # class_weight
    #-------------------------------
    # SVC
    [0.1, 0.5, 1,2,3,4,5,7,9,10,12,15,20,30],       # C(7)
    [2,5,10],                                       # degree(for poly)
    ['scale', 'auto'],                              # gamma
    [0,1,2,3,4, 5,7,9,11,15, 20],                   # coef0 https://zhuanlan.zhihu.com/p/672562238
    [True, False],                                  # shrinking
    [True, False],                                  # probability
    ['balanced', 'None'],                           # class_weight（public）
    ['ovo', 'ovr'],                                 # decision function shape
    [True, False]                                   # break_ties
]

# XGBoost 在 SKlearn中的接口
xgb_sk_params = [
    # 模型/弱分类器选择：XGB XGBRF 
    [0, 1],                                                     # 0 : XGBClassifier 1 : XGBRFClassifier
    ['gbtree', 'gblinear','dart'],                              # booster
    #-----------------------
    [10],                                                       # n_estimator
    # public
    [0, 0.5, 1],                                                # reg_alpha (3)
    [0, 0.5, 1],                                                # reg_lambda
    ['multi:softmax', 'binary:logistic','reg:squarederror'],                       # objective 损失函数
    [0.3, 0.5, 0.7],                                                            # base_score
    [0],                                                        # max_delta_step    0为无约束 https://www.cnblogs.com/TimVerion/p/11436001.html#:~:text=max_delta_step%EF%BC%9A%E9%BB%98%E8%AE%A40%2C%E6%88%91%E4%BB%AC%E5%B8%B8%E7%94%A80.,%E8%BF%99%E4%B8%AA%E5%8F%82%E6%95%B0%E9%99%90%E5%88%B6%E4%BA%86%E6%AF%8F%E6%A3%B5%E6%A0%91%E6%9D%83%E9%87%8D%E6%94%B9%E5%8F%98%E7%9A%84%E6%9C%80%E5%A4%A7%E6%AD%A5%E9%95%BF%EF%BC%8C%E5%A6%82%E6%9E%9C%E8%BF%99%E4%B8%AA%E5%8F%82%E6%95%B0%E7%9A%84%E5%80%BC%E4%B8%BA0%2C%E5%88%99%E6%84%8F%E5%91%B3%E7%9D%80%E6%B2%A1%E6%9C%89%E7%BA%A6%E6%9D%9F%E3%80%82
    # ['gain','weight','cover','total_gain','total_cover'],     # importance_type
    # ['logloss', 'mlogloss'],                                  # eval_metric
    # gbtree dart
    [3, 6, 12],                                                 # max_depth (8)
    [0.01, 0.1,0.3],                                            # learning_rate
    [0, 1],                                                     # gamma
    [1, 5, 10],                                                 # min_child_weight
    [1],                                                        # subsample 采样比例
    [1],                                                        # colsample_bytree colsample_bylevel colsample_bynode # 使用1
    ['auto'],                                                   # tree_method
    # dart
    ['tree', 'forest'],                                         # normalize_type(15)
    # gblinear
    ['shotgun'],                                                # updater (16) 下降算法 （默认值）    https://zhuanlan.zhihu.com/p/687910054
    ['cyclic','random','greedy'],                               # feature_selector
    # nthread默认为-1（使用所有线程）
]

# 单独手动调参(基于pytorch的强分类MLP)
mlp_torch_param =[
        [4],        # input_dim         
        [50],
        [100],
        [50],
        [1],        # output_dim 4
        [512],
        [512],
        [512],
        [512],
        [512], 
        [1],        # output_dim 10
        [0],        # criterion 0 : CrossEntropyLoss
        [0,1]       # optimizer 0 : Adam 1.SGD(默认使用Adam)
    ]

